### DISCUSSION TITLE
How to run tests with Bun?

### DISCUSSION THREAD

**Discussion by Extarys** [vitest-dev/vscode#473] [Q&A] [answered]
Hey there,

Is there a way to run the tests with bun instead of node ? I'd like to use `Bun.` functions in my tests.

I never really did tests before, maybe this needs to be configured inside `vite.config.ts` ?

**Comment by sheremet-va** [comment, marked as answer]
Bun doesn't support some essential APIs, so you can't do that: https://github.com/oven-sh/bun/issues/4145

**Reply by Extarys** [reply]
Oh I see, thank you. I'll find some workaround for the Bun. methods I intended to use :)

**Reply by bauchdj** [reply]
Is this still the case? I can run all my tests with bun successfully. `bun run --bun vitest`

**Reply by boredland** [reply]
@bauchdj interestingly, when I try this, I am getting lots of errors:

```
TypeError: null is not an object (evaluating 'this._thread.stdout.pipe')
 ❯ start node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7732:16
 ❯ start node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7723:16
 ❯ start node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7463:23
 ❯ start node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7453:14
 ❯ schedule node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:8018:18
 ❯ schedule node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7983:19
 ❯ run node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7980:8
 ❯ run node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:7969:12
 ❯ <anonymous> node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:8220:17
 ❯ <anonymous> node_modules/vitest/dist/chunks/cli-api.BKg19Fvw.js:8217:38
```

**Reply by bauchdj** [reply]
@boredland Without more information I don't know why. I switched to `provider: istanbul` for coverage. I believe that was the only change I had to make to my vitest config.

**Reply by N0SAFE** [reply]
in the issue https://github.com/oven-sh/bun/issues/4145 its said that this as been fixed and bun now fully support vitest including the coverage. check the blog at https://bun.com/blog/bun-v1.4#vitest
